contracts/examples/LiquidityExamples.sol

```diff
@@ -0,0 +1,249 @@
+                tokenId: tokenId,
+                recipient: address(this),
+                amount0Max: type(uint128).max,
+                amount1Max: type(uint128).max
+            });
+
+        (amount0, amount1) = nonfungiblePositionManager.collect(params);
+
+        // send collected feed back to owner
+        _sendToOwner(tokenId, amount0, amount1);
+    }
+
+    /// @notice A function that decreases the current liquidity by half. An example to show how to call the `decreaseLiquidity` function defined in periphery.
+    /// @param tokenId The id of the erc721 token
+    /// @return amount0 The amount received back in token0
+    /// @return amount1 The amount returned back in token1
+    function decreaseLiquidityInHalf(uint256 tokenId) external returns (uint256 amount0, uint256 amount1) {
+        // caller must be the owner of the NFT
+        require(msg.sender == deposits[tokenId].owner, 'Not the owner');
+        // get liquidity data for tokenId
+        uint128 liquidity = deposits[tokenId].liquidity;
+        uint128 halfLiquidity = liquidity / 2;
+
+        // amount0Min and amount1Min are price slippage checks
+        // if the amount received after burning is not greater than these minimums, transaction will fail
+        INonfungiblePositionManager.DecreaseLiquidityParams memory params =
+            INonfungiblePositionManager.DecreaseLiquidityParams({
+                tokenId: tokenId,
+                liquidity: halfLiquidity,
+                amount0Min: 0,
+                amount1Min: 0,
+                deadline: block.timestamp
+            });
+
+        (amount0, amount1) = nonfungiblePositionManager.decreaseLiquidity(params);
+
+        //send liquidity back to owner
+        _sendToOwner(tokenId, amount0, amount1);
+    }
+
+    /// @notice Increases liquidity in the current range
+    /// @dev Pool must be initialized already to add liquidity
+    /// @param tokenId The id of the erc721 token
+    /// @param amount0 The amount to add of token0
+    /// @param amount1 The amount to add of token1
+    function increaseLiquidityCurrentRange(
+        uint256 tokenId,
+        uint256 amountAdd0,
+        uint256 amountAdd1
+    )
+        external
+        returns (
+            uint128 liquidity,
+            uint256 amount0,
+            uint256 amount1,
+            IUniswapV3Pool pool
+        )
+    {
+        AddLiquidityParams memory params =
+            AddLiquidityParams({
+                token0: deposits[tokenId].token0,
+                token1: deposits[tokenId].token1,
+                fee: poolFee,
+                recipient: address(this),
+                tickLower: deposits[tokenId].tickLower,
+                tickUpper: deposits[tokenId].tickUpper,
+                amount0Desired: amountAdd0,
+                amount1Desired: amountAdd1,
+                amount0Min: 0,
+                amount1Min: 0
+            });
+
+        (liquidity, amount0, amount1, pool) = addLiquidity(params);
```

Comment:
take a look at where the nftManager uses`addLiquidity`. It does a lot of things to update the position. I think you're better off calling the nft manager directly for this: `nftManager.increaseLiquidity()`

in this case you'd be adding liquidity on behalf of this contract, not behalf of an nft.